packages/starlight/user-components/rehype-file-tree.ts

```diff
@@ -0,0 +1,245 @@
+
+			if (isDirectory) {
+				const hasContents = otherChildren.length > 0;
+
+				node.children = [
+					h('details', { open: hasContents }, [
+						h('summary', treeEntry),
+						...(hasContents ? otherChildren : [h('ul', h('li', '…'))]),
+					]),
+				];
+
+				// Continue down the tree.
+				return CONTINUE;
+			}
+
+			node.children = [treeEntry, ...otherChildren];
+
+			// Files can’t contain further files or directories, so skip iterating children.
+			return SKIP;
+		});
+	};
+});
+
+/** Make a text node with the pass string as its contents. */
+function makeText(value = ''): Text {
+	return { type: 'text', value };
+}
+
+/** Make a node containing an SVG icon from the passed HTML string. */
+function makeSVGIcon(svgString: string) {
+	return s(
+		'svg',
+		{
+			width: 16,
+			height: 16,
+			class: 'tree-icon',
+			'aria-hidden': 'true',
+			viewBox: '0 0 24 24',
+		},
+		fromHtml(svgString, { fragment: true })
+	);
+}
+
+/** Return the icon for a file based on its file name. */
+function getFileIcon(fileName: string) {
+	const name = getFileIconName(fileName);
+	if (!name) return defaultFileIcon;
+	if (name in Icons) {
+		const path = Icons[name as keyof typeof Icons];
+		return makeSVGIcon(path);
+	}
+	return defaultFileIcon;
+}
+
+/** Return the icon name for a file based on its file name. */
+function getFileIconName(fileName: string) {
+	let icon = definitions.files[fileName];
+	if (icon) return icon;
+	icon = getFileIconTypeFromExtension(fileName);
+	if (icon) return icon;
+	for (const [partial, partialIcon] of Object.entries(definitions.partials)) {
+		if (fileName.includes(partial)) return partialIcon;
+	}
+	return icon;
+}
+
+/**
+ * Get an icon from a file name based on its extension.
+ * Note that an extension in Seti is everything after a dot, so `README.md` would be `.md` and
+ * `name.with.dots` will try to look for an icon for `.with.dots` and then `.dots` if the first one
+ * is not found.
+ */
+function getFileIconTypeFromExtension(fileName: string) {
+	const firstDotIndex = fileName.indexOf('.');
+	if (firstDotIndex === -1) return;
+	let extension = fileName.slice(firstDotIndex);
+	while (extension !== '') {
+		const icon = definitions.extensions[extension];
+		if (icon) return icon;
+		const nextDotIndex = extension.indexOf('.', 1);
+		if (nextDotIndex === -1) return;
+		extension = extension.slice(nextDotIndex);
+	}
+	return;
+}
+
+/** Validate that the user provided HTML for a file tree is valid. */
+function validateFileTree(tree: Element) {
+	const rootElements = selectAll('body > *', tree);
+	const [rootElement] = rootElements;
+
+	if (rootElements.length === 0) {
+		throwFileTreeValidationError(
+			`The <FileTree> component expects its content to be a unique unordered list but found no elements.`
+		);
+	}
+
+	if (rootElements.length !== 1) {
+		throwFileTreeValidationError(
+			`The <FileTree> component expects its content to be a unique unordered list but found the following elements: ${rootElements
```

Comment:
Language nit: “unique” in English suggests the list is different from any other list rather than the French “unique” meaning singular. Maybe also helps to repeat that the issue is “multiple” children?

```suggestion
			`The <FileTree> component expects its content to be a single unordered list but found multiple child elements: ${rootElements
```